src/include/mpir_request.h

```diff
@@ -215,6 +215,7 @@ struct MPIR_Request {
 #endif                          /* HAVE_DEBUGGER_SUPPORT */
             /* Persistent requests have their own "real" requests */
             struct MPIR_Request *real_request;
+            MPIR_TSP_sched_t sched;
```

Comment:
Can't review this until the use case -- the composition of schedulers -- is explained and understood.

Reply:
This commit basically adds a new type of vertex in tsp, which is another scheduler. So this basically allows recursively adding schedulers to an existing schedule. Let me know if this general explanation is  acceptable. If not, I will see I will remove this commit for now.